w5500/w5500.go

```diff
@@ -0,0 +1,262 @@
+package w5500
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"sync"
+
+	"tinygo.org/x/drivers"
+	"tinygo.org/x/drivers/netdev"
+)
+
+var _ netdev.Netdever = &Device{}
+
+var _debug debug = debugOff
+
+// Resolver is a function that resolves a hostname to an IP address.
+type Resolver func(host string) (netip.Addr, error)
+
+// PinOutput is a function that sets a pin high or low.
+type PinOutput func(level bool)
+
+// Device is a driver for the W5500 Ethernet controller.
+type Device struct {
+	maxSockets  int
+	maxSockSize int
+
+	mu  sync.Mutex
+	bus drivers.SPI
+	cs  PinOutput
+	dns Resolver
+
+	sockets []*socket
+	laddr   netip.Addr
+
+	cmdBuf []byte
+}
+
+// New returns a new w5500 driver.
+func New() *Device {
+	return &Device{
+		cmdBuf: make([]byte, 3),
+	}
+}
+
+// Config is the configuration for the device.
+//
+// The SPI bus must be fully configured.
+type Config struct {
+	SPI drivers.SPI
```

Comment:
I believe we usually pass things like `drivers.SPI` and pins into `New()` method, instead of having them in Config struct.